### PR TITLE
Add GDScript filetype and use BufNewFile,BufRead

### PR DESCRIPTION
**Pull request by Shatur95**
I used `BufNewFile,BufRead` as suggested in `:h ftdetect`. I also noticed that we have `query.vim` that use `BufEnter,BufNewFile`. Probably we should change it to `BufNewFile,BufRead` too.

### CODE REVIEW

**Comment by vigoux** [comment]
> I used `BufNewFile,BufRead` as suggested in `:h ftdetect`. I also noticed that we have `query.vim` that use `BufEnter,BufNewFile`. Probably we should change it to `BufNewFile,BufRead` too.

I agree on changing the query filetype too to the suggested values.

**Review by vigoux — APPROVED** [review]

**Comment by Shatur95** [comment]
@vigoux, pushed!

File: ftdetect/gdscript.vim [hunk 1]
```diff
@@ -0,0 +1 @@
+autocmd BufNewFile,BufRead *.gd set ft=gdscript
```
Comment (theHamsta) [inline ftdetect/gdscript.vim:1]: I'm wondering whether we should do this for the other filetypes that aren't supported by nvim too?

Comment (Shatur95) [reply ftdetect/gdscript.vim:1]: I think so.

Comment (theHamsta) [reply ftdetect/gdscript.vim:1]: fish, svelte, maybe julia?

**Pull request merged**